### PR TITLE
 Migrate the cluster registry design and plan to the community repo

### PR DESCRIPTION
**Pull request by perotinus**
/cc @csbell @quinton-hoole 

This is a mechanical conversion of the [Cluster registry project plan doc](https://docs.google.com/document/d/1bVvq9lDIbE-Glyr6GkSGWYkLb2cCNk9bR8LL7Wm-L6g/edit), with a few tweaks to better suit the format and to note the provenance of this document.

Quinton and Christian, I'd like both of you to sign-off on this before I commit. Anyone else is free to chime in if they have concerns, but I believe that we achieved consensus on this doc, and I'd like to formalize that.

@font @pmorie @ericchiang

### CODE REVIEW

File: contributors/design-proposals/multicluster/cluster-registry/project-design-and-plan.md [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+-   Supports independent authz for reading/writing clusters
+-   `kubectl` integration with the cluster registry API server is first-class
+    and on par with kubectl integration with core Kubernetes APIs
+-   Supports grouping of clusters
+-   Supports specifying and enforcing read and/or write authorization for groups
+    of clusters in the registry
+-   Working Federation integration
+-   Supports status from active controllers
+
+### Later {#later}
```
Comment (csbell) [inline contributors/design-proposals/multicluster/cluster-registry/project-design-and-plan.md]: {#later} ?

Comment (perotinus) [reply contributors/design-proposals/multicluster/cluster-registry/project-design-and-plan.md]: Whoops! That's an artifact from the markdown conversion. Thanks for catching that!

**Comment by perotinus** [comment]
Thanks for the review, @csbell!

**Comment by perotinus** [comment]
@quinton-hoole ping?

**Comment by perotinus** [comment]
/cc @quinton-hoole 
I'm planning to submit this this evening, unless I hear otherwise from you.

**Comment by csbell** [comment]
Looks good.
/lgtm

**Pull request merged**